src/test/test_http_client_builder.rs

```diff
@@ -0,0 +1,12 @@
+use reqwest::{
+    header::{self, HeaderMap, HeaderValue},
+    ClientBuilder,
+};
+
+/// Create a http client builder that with some default headers
```

Comment:
Can you add a one or more sentences to the functions documentation that explain why we set a custom header? I think that would be really helpful when someone looks at this in the future. Something inspired by the issue, maybe along the lines of:

> Create a pre-configured ClientBuilder
>
> This function returns a `reqwest::ClientBuilder` that has been pre-configured with default headers. Specifically, it sets the `User-Agent` header so that requests from the test suite can be more easily filtered and inspected in the request logs.